File: backend/loocks/src/main/java/pin/loocks/domain/dtos/ClothingAnalysisDTO.java
package pin.loocks.domain.dtos;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.Data;
import pin.loocks.domain.enums.Estacion;
import pin.loocks.domain.enums.Estilo;
import pin.loocks.domain.enums.Zona;
import pin.loocks.domain.models.PorcentajeColor;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ClothingAnalysisDTO {
	private String nombre;
	private String marca;

	private String colorPrimario;
	private List<PorcentajeColor> colores;

	private Estilo estilo;
	private Estacion estacion;
	private List<Zona> zonasCubiertas;
	private Boolean puedePonerseEncimaDeOtraPrenda;

	private Double nivelDeAbrigo;
	private String base64Img = null;
}

File: mobile/lib/domain/dtos/clothing-analysis-dto.ts
import { Estacion } from '../enums/estacion';
import { Estilo } from '../enums/estilo';
import { TipoArticulo } from '../enums/tipo-articulo';

export interface ClothingAnalysisDTO {
  nombre: string;
  marca?: string;
  colores: ColorInfo[];
  colorPrimario: string;
  estacion: Estacion;
  tipo: TipoArticulo;
  base64Img?: string;
  estilo?: Estilo;
}

export interface ColorInfo {
  color: string;
  porcentaje: number;
}
